packages/block/src/header.ts

```diff
@@ -208,6 +218,12 @@ export class BlockHeader {
       this._common.setHardforkByBlockNumber(number.toNumber())
     }
 
+    if (this._common.isActivatedEIP(1559) && baseFeePerGas === undefined) {
+      this.baseFeePerGas = new BN(1)
+    } else {
+      this.baseFeePerGas = undefined
```

Comment:
Yes, I added this when refactoring out the public `BlockHeader.getCommon()` class in 502b9b2, since the solution before needed a final `Common` instance before instantiation which has stronger side effects and is not desirable. The exact length check was then not possible any more and this setting to `undefined` was therefore for the case where a `baseFeePerGas` was passed in for a non-1559 block.

I will update though with a check to just throw here in this case, this is likely a lot better than implicitly overwrite. And then we completely match the functionality we had before with Jochem's implementation without the `Common` side effects.

(will do on next push).